src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -217,9 +217,19 @@ static smt_termt convert_expr_to_smt(const bitxor_exprt &bitwise_xor)
 
 static smt_termt convert_expr_to_smt(const bitnot_exprt &bitwise_not)
 {
-  UNIMPLEMENTED_FEATURE(
-    "Generation of SMT formula for bitwise not expression: " +
-    bitwise_not.pretty());
+  const bool operand_is_bitvector =
+    can_cast_type<integer_bitvector_typet>(bitwise_not.op().type());
+
+  if(operand_is_bitvector)
+  {
+    return smt_bit_vector_theoryt::make_not(
+      convert_expr_to_smt(bitwise_not.op()));
```

Comment:
Random place for a comment. This commits title doesn't display properly in git hubs UI because it is too long. The fact that is implementation and tests isn't needed in the title. So I'd reduce it down to something like "Add conversion of `bitnot_exprt` to SMT terms"